.travis.yml

```diff
@@ -1,6 +1,9 @@
 language: java
 jdk:
-  - openjdk8
+  - openjdk11
```

Comment:
This a massive change here! If we compile in Tycho with Java 11, at the very least, there may be issues with some Java standard library entries not being matched (e.g. if somebody uses a method that is only available in Java 11, it may compile but fail with runtime exceptions when using Java 8); in the worst, even the compiled bytecode may not work with Java 8. This can be worked around by the correct maven toolchain setup, but I don't recall that part being added to the build configuration.

Reply:
This is just for build environment on Travis and it should not impact anything on build target.  OpenJDK11 is required by Tycho 2.3.0, and I do not think the latest Eclipse environment supports any JDK version older than 11.  Build source and target are controlled by maven the following configurations of pom.xml as
` <maven.compiler.target>1.8</maven.compiler.target>`
`<maven.compiler.source>1.8</maven.compiler.source>`
These are not change.

Reply:
Regarding MANIFEST.MF,  each bundle should properly specify which runtime environment is supported.  It should not be updated by the change of build environment because  we assume or trust JRE keeps backward compatibility.

Reply:
This setting is used just for the build environment and my issue is that it is not enough. I think (but I am not sure so it should be tested), the mentioned settings would cause the compiler to emit Java 8 specific byte code, so for now, it should work.

However, considering both the development environments (required by current Eclipse versions) and the build system uses Java 11 while we state that the built bundles work in Java 8, who will catch issues as somebody using newer standard library code (e.g. [List.of](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html#of(E)) that got introduced in Java 9), it would compile fine but if somebody tried to use the bundles in a Java 8 environment the built-in checks in Eclipse would not flag it. If we are going to support only Java 11, why not state it explicitly in the code as well?

Reply:
Your concern that the runtime may mismatch with what we assume should be handled with `Bundle-RequiredExecutionEnvironment` of OSGi configuration in MANIFEST.MF, as you probably already noticed.

Each bundle we develop can still support Java8 by specifying JavaSE-1.8 in Bundle-RequiredExecutionEnvironment so that bundle users may use it with JRE8.   Practically it will be harder and harder because such bundle will depend on and be tested with the latest Eclipse bundles that support only Java11 or later.   However, that will not be a good reason to specify Java11 in Bundle-RequiredExecutionEnvironment because it's better to allow developers to port or use such bundles in different environment and it would be simply harmful in view of portability to specify Java11 if they do not use Java11 specific features.

Anyway, that is independent of the build environment we use.   We expect JDK keeps backward compatibility with the compiled bytecode.  So when the development environment, in this case, Tycho 2.3.0, requires JDK11 feature, we need to use JDK11 anyway.

Of course what runtime environment we officially support is another story.  It involves many other factors such as testing, deployment, and support.  We should discuss it separately.

Reply:
If this requirement specification will be done in relation to [ST6RI-383](https://openmbee.atlassian.net/browse/ST6RI-383) I am fine with postponing this until that point, but I really believe it should be done as soon as possible.

Reply:
@ujhelyiz, as I said, I disagree to change the bundles' Bundle-RequiredExecutionEnvironment for this reasons.

Reply:
I understand your reasoning, as you are saying you want to build system to keep the built java classes compatible with Java 8, so not changing the BREE values makes sense.

However, my point is that the build system currently does not guarantee that everything will work in Java 8 when built on Java 11 as classes and methods of the JDK running the build might be resolved in a way that is not compatible with Java 8 (I am speaking from experience, such issues are really hard to find out). However, given that it is planned to update minimum requirement in a different ticket in the not too distant future, it might make sense to ignore this issue for now.

Reply:
Indeed, I agree on your point.  Also in my experience, changing runtime environment made unexpected problems and it's really hard to find.  For us, I consumed almost one week about the conflicts of OSGi and runtime environment and adding only one single option of target environment solved the problems ;)  Lessons learned is we need ample time after the migration of runtime and build environment.